loader/include/gz/plugin/Loader.hh

```diff
@@ -133,6 +133,18 @@ namespace gz
       public: std::unordered_set<std::string> LoadLib(
                   const std::string &_pathToLibrary);
 
+
+      /// \brief Load a library at the given path
+      ///
+      /// \param[in] _pathToLibrary
+      ///   The path to a library
+      /// \param[in] _noDelete
+      ///   If true, RTLD_NODELETE will be used when loading the library.
+      ///
+      /// \returns The set of plugins that have been loaded from the library
+      public: std::unordered_set<std::string> LoadLib(
+                  const std::string &_pathToLibrary, bool _noDelete);
```

Comment:
How about supporting any arbitrary `dlopen` flags here instead of just a boolean? This should make it easier to try out different flags as we need them. I started something like that on

* https://github.com/gazebosim/gz-plugin/pull/35

Reply:
As I mentioned on slack, `RTLD_NODELETE` is not a POSIX compliant flag, and thus our windows version of `dlopen` (via dlfcn-32` doesn't have this flag. So the logic to detect MSVC has to be spread everywhere `LoadLib` is called. I'd prefer to keep this as a separate flag and keep the MSVC detection logic in this library.